### PR TITLE
Add AzureSasCredential

### PR DESCRIPTION
**Pull request by kasobol-msft**
**In this PR:**
- Add `AzureSasCredential` per https://github.com/Azure/azure-sdk/issues/1954
- `AzureSasCredential` is the name that has been settled on the end of discussion.
- Add `AzureSasCredentialPolicy` that appends SAS to query

**Remarks:**
- Some service (like storage in the Portal) present SAS with leading "?". This has to be stripped before appending
- The validation if serviceUri already contain sas (mentioned [here](https://github.com/Azure/azure-sdk/issues/1954#issuecomment-718190611)) will be responsibility of service clients:
    - the format varies between services (i.e. Event Grid SAS and Storage SAS are vastly different)
    - it would be good to fail fast (at client creation) rather than late (at request send).

**References**
- [.NET PR](https://github.com/Azure/azure-sdk-for-net/pull/17636)

### CODE REVIEW

**Comment by annatisch** [comment]
Thanks @kasobol-msft - looks good.
Could you please update release notes? I assume this will initially be going out as a preview version?

**Comment by kasobol-msft** [comment]
> Thanks @kasobol-msft - looks good.
> Could you please update release notes? I assume this will initially be going out as a preview version?

As discussed offline. We intend to ship this in GA. I've added release notes entry.

**Comment by xiangyan99** [comment]
What's the difference betweeen AzureSasCredential and AzureKeyCredential? Can we just create an alias of AzureKeyCredential?

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@
 
 ## 1.9.1 (Unreleased)
```
Comment (xiangyan99) [inline sdk/core/azure-core/CHANGELOG.md:4]: Given this is a new feature, it should be 1.10.0

Comment (kasobol-msft) [reply sdk/core/azure-core/CHANGELOG.md]: fixed.

**Comment by kasobol-msft** [comment]
> What's the difference betweeen AzureSasCredential and AzureKeyCredential? Can we just create an alias of AzureKeyCredential?

There isn't much today. However per https://github.com/Azure/azure-sdk/issues/1954 the AzureKeyCredential will get extra property (name). The outcome from board discussion was to keep these types separate.

**Review by annatisch — APPROVED** [review]

**Pull request merged**